lib/spack/spack/test/config.py

```diff
@@ -857,3 +869,67 @@ def test_dotkit_in_config_does_not_raise(
     # we throw a a deprecation warning without raising
     assert '_sp_sys_type' in captured[0]  # stdout
     assert 'Warning' in captured[1]  # stderr
+
+
+def test_internal_config_section_override(mock_low_high_config,
+                                          write_config_file):
+    write_config_file('config', config_merge_list, 'low')
+    wanted_list = config_override_list['config']['build_stage:']
+    mock_low_high_config.push_scope(spack.config.InternalConfigScope
+                                    ('high', {
+                                        'config:': {
+                                            'build_stage': wanted_list
+                                        }
+                                    }))
+    assert wanted_list == mock_low_high_config.get('config:build_stage')
+
+
+def test_internal_config_dict_override(mock_low_high_config,
+                                       write_config_file):
+    write_config_file('config', config_merge_dict, 'low')
+    wanted_dict = config_override_dict['config']['info:']
+    mock_low_high_config.push_scope(spack.config.InternalConfigScope
+                                    ('high', config_override_dict))
+    assert wanted_dict == mock_low_high_config.get('config:info')
+
+
+def test_internal_config_list_override(mock_low_high_config,
```

Comment:
This feels redundant with `test_set_list_override`. I prefer `test_set_list_override` because it goes through the `config.override` function which is a higher-level part of the API.

Reply:
`test_internal_config_list_override()` was added to complete the existing set of `test_internal_config...()` functions after adding the new functionality. I added the `test_set_...() functions because I believed they test a sufficiently different code path. I'm not sure it makes sense to retire one `test_internal_config...()` test as redundant without retiring all the others for the same reason. Are you sure all these tests don't exercise enough either distinct or separable low level code to justify their existence?

Reply:
My original thought was that the overall functionality we want to test is whether we can use `spack.config.override` and that is agnostic to implementation details like whether we use an `InternalConfigScope` underneath to achieve that.

However, it is true that `InternalConfigScope` should be able to handle overrides regardless of whether it is used by `spack.config.override`. Therefore I think this is worthwhile to test.